### PR TITLE
add cities lookup V2

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

This adds a new cities lookup CSV which uses [ISO_3166-2](https://en.wikipedia.org/wiki/ISO_3166-2) for region and country codes. This is because Fastly has deprecated it's old `geoip` API in favour of it's [`client.geo`](https://github.com/guardian/fastly-edge-cache/issues/942) APIs.

The latitude and longitude were added from the [geonames cities1000 dump](http://download.geonames.org/export/dump/). This has left us with less coverage (`161360` cities vs `353225`), bu8t still covers cities with a population > 1000, and we [fall back to text search](https://github.com/guardian/frontend/blob/main/onward/app/weather/controllers/LocationsController.scala#L63-L71) via AccuWeather if there isn't a Lat/Long match. 

I'd like to write some tests around this to help the next person who has to go through this, but we're racing against the clock a little (EOD today the APIs are deprecated). The current tests just make sure the lookups work when given any data.

For a little confidence I have created some dashboards on how many matches / misses we're getting from the current system, I would hope to have these stay about the same. They're based on [hit](https://github.com/guardian/frontend/blob/main/onward/app/weather/controllers/LocationsController.scala#L45) / [miss](https://github.com/guardian/frontend/blob/main/onward/app/weather/controllers/LocationsController.scala#L62) logs from the LocationsController.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

This relies on [this PR](https://github.com/guardian/fastly-edge-cache/pull/941/files) being merged.

### CODE REVIEW

**Review by OllysCoding — APPROVED** [review]
+1 Looks good to me

**Comment by OllysCoding** [comment]
My understanding is that this PR just adds the database & its implementation in, but it isn't being consumed anywhere just yet?

File: onward/app/weather/geo/CitiesLookUpV2.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  }
+}
+
+case class CitiesCsvLine(
+    geonamesId: Int,
+    city: String,
+    region: String,
+    country: String,
+    latitude: Double,
+    longitude: Double,
```
Comment (mxdvl) [inline onward/app/weather/geo/CitiesLookUpV2.scala:52]: I think a `Float` would be plenty, here.

File: onward/app/weather/geo/CitiesLookUpV2.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    Source.fromInputStream(getInputStream("city_region_country_lat_long.csv").get, "iso-8859-1")
+  }
+
+  private[geo] def getCsvLines: Iterator[CitiesCsvLine] = {
+    val csv = getGeoIPCityInputStream
+    csv.getLines().drop(2) collect { case CitiesCsvLine(csvLine) => csvLine }
+  }
+
+  def loadCache(): Map[CityRef, LatitudeLongitude] = {
+    // first two lines are Copyright info and field names
```
Comment (mxdvl) [inline onward/app/weather/geo/CitiesLookUpV2.scala:66]: Is this comment meant to live before the `drop(2)`?

**Review by mxdvl — APPROVED** [review]
I like [how simple reading a CSV is](https://alvinalexander.com/scala/csv-file-how-to-process-open-read-parse-in-scala/)!

I’m happy for this to go in as-is, due to time pressure, but would love to have some tests added in the near future.

File: onward/app/weather/geo/CitiesLookUpV2.scala (hunk 2 above)
Comment (jamesgorrie) [reply onward/app/weather/geo/CitiesLookUpV2.scala]: Nice catch - it should also be `drop(1)` as we don't have copyright info in there.

File: onward/app/weather/geo/CitiesLookUpV2.scala (hunk 1 above)
Comment (mxdvl) [reply onward/app/weather/geo/CitiesLookUpV2.scala:52]: Just thought it was worth knowing that 6 decimal places gives you 1m precision.

https://gis.stackexchange.com/a/2964/95613

**Comment by jamesgorrie** [comment]
@OllysCoding that was the case, I have updated this to actually implement the new lookup.

**Comment by jamesgorrie** [comment]
Closed in favour of #25033

**Pull request closed without merging**